kernel/inductive.ml

```diff
@@ -913,7 +913,7 @@ let rec subterm_specif renv stack t =
         let decrArg = recindxs.(i) in
         let theBody = bodies.(i)   in
         let nbOfAbst = decrArg+1 in
-        let sign,strippedBody = Term.decompose_lam_n_assum nbOfAbst theBody in
+        let sign,strippedBody = dest_lam_n_assum renv.env nbOfAbst theBody in
                    (* pushing the fix parameters *)
```

Comment:
My assumption is that recindxs only counts assumptions, not definitions.

Reply:
Yes, thus a `dest_lam_n_assum` not counting the lets (i.e. with `n` and not `n-1` in the let case) is ok, right?

Reply:
Yes, that's better :)